src/main/java/com/gmail/goosius/siegewar/utils/SiegeWarDominationAwardsUtil.java

```diff
@@ -0,0 +1,266 @@
+            nations = cullNationsWithTooFewDominationRecords(nations);
+            if(nations.size() == 0) {
+                SiegeWar.info("Global Domination Awards: No nations qualified for awards this week, due to having too few domination records.");
+                return; 
+            }
+
+            //The number of awardees will be as configured, or the size of the nations list, whichever is smaller, 
+            int numberOfAwardees = Math.min(moneyToGrant.size(), nations.size());
+
+            //Sort nations by recorded dominance
+            nations.sort(SiegeWarNationUtil.BY_GLOBAL_DOMINATION_RANKING);     
+                  
+            //Gib awards
+            Nation nation = null;
+            for(int nationPosition = 0; nationPosition < numberOfAwardees; nationPosition++) {
+                try{
+                    nation = nations.get(nationPosition);            
+                    //Gib money
+                    nation.getAccount().deposit(moneyToGrant.get(nationPosition), "Global Domination Award");    
+                    //Gib artifacts
+                    grantArtefactsToNation(artefactsToGrant.get(nationPosition), nation);
+                    //Global message
+                    System.out.println("Global Domination Awards Granted");
+                } catch(Throwable t) {
+                    SiegeWar.severe("Problem granting global domination award to nation " + nation.getName());
+                    SiegeWar.severe(t.getMessage());
+                    t.printStackTrace();
+                }
+            }
+
+            //Remove all domination records
+            nations = new ArrayList<>(TownyUniverse.getInstance().getNations());
+            for(Nation nationForRecordRemoval: nations) {
+                NationMetaDataController.setDominationRecord(nationForRecordRemoval, Collections.emptyList());
+                nationForRecordRemoval.save();
+            }
+        }       
+    }
+
+    private static List<Nation> cullNationsWithTooFewDominationRecords(List<Nation> nations) {
+        int requiredRecords = SiegeWarSettings.getDominationAwardsGlobalMinimumAssessmentPeriodHours();
+        List<Nation> result = new ArrayList<>();
+        for(Nation nation: nations) {
+            if(NationMetaDataController.getDominationRecord(nation).size() >= requiredRecords)
+                result.add(nation);
+        }
+        return result;
+    }
+
+    private static void grantArtefactsToNation(List<Integer> offersToGrantFromEachTier, Nation nation) {
+        List<ItemStack> artefactsToGrant = new ArrayList<>();
+        for(int tier = 0; tier < offersToGrantFromEachTier.size(); tier++) {
+            artefactsToGrant.addAll(generateArtefacts(tier, offersToGrantFromEachTier.get(tier)));
+        }
+        //Get chunk of capital homeblock
+        WorldCoord homeBlockCoord = nation.getCapital().getHomeBlockOrNull().getWorldCoord();
```

Comment:
This should really test if the homeblock is null before getting the worldCoord(), in a perfect world there's no issues but an admin changing a world's name or otherwise messing with the townblocks folder can result in the homeblock/spawn not being set.

Reply:
Done in next push